libfreerdp/core/peer.c

```diff
@@ -455,8 +455,10 @@ static int peer_recv_callback(rdpTransport* transport, wStream* s, void* extra)
 	switch (rdp->state)
 	{
 		case CONNECTION_STATE_INITIAL:
-			if (!rdp_server_accept_nego(rdp, s))
+			if (!rdp_server_accept_nego(rdp, s)) {
+				WLog_DBG(TAG, "peer_recv_callback: CONNECTION_STATE_INITIAL - rdp_server_accept_nego() fail");
 				return -1;
```

Comment:
I feel like this log will be very verbose when you connect to a RDP security only server, with 3 connection attempts (NLA, then TLS, and finally RDP)...